### DISCUSSION TITLE
Github API to get all public repos fro mall users

### DISCUSSION THREAD

**Discussion by tammytdo** [community/community#119379] [Apps, API and Webhooks] [answered]
### Select Topic Area

General

### Body

I’m working on a project and want to pull data of all github repos with a contribution in the last month (and filter just the top 100 so that it’s not such a massive amount of data). Any pointers of weather an endpoint for this exists?

**Comment by borderss** [comment, marked as answer]
You could try using the github REST API, specifically filtering for all repositories with query parameter "pushed" after some date.
https://docs.github.com/en/search-github/getting-started-with-searching-on-github/understanding-the-search-syntax
https://docs.github.com/en/rest/search/search

Apparently each search is limited to up to 1,000 results for each search, so should be fine for your use case.

**Reply by tammytdo** [reply]
Hi! This looks like it can only search for one user at a time. The search query is exactly what I need. I'd just like to find every single user and every single repo and then apply the query.

**Reply by borderss** [reply]
If you query for repositories with query parmeter "pushed" after some date, it will return up to 1000 entries of all repositories that had pushes after that date.

So esentially you get the dataset you are looking for, without having to get all users beforehand.

**Reply by tammytdo** [reply]
Ohhh I see what you're saying. This is perfect. Thank you!
